### ISSUE TITLE
[BUG/Help] stream_chat 无法输出 emoji  会变成�

### ISSUE THREAD

**Issue by wuyun1** [THUDM/ChatGLM-6B#478]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

代码：

```
from transformers import AutoTokenizer, AutoModel

tokenizer = AutoTokenizer.from_pretrained("THUDM/chatglm-6b-int4-qe", trust_remote_code=True) #
model = AutoModel.from_pretrained("THUDM/chatglm-6b-int4-qe",trust_remote_code=True).float().eval()

prompt = "你知道笑脸的 emoji 是哪个吗？ Unicode编码是多少？"
output_pos = 0
for response, history in model.stream_chat(tokenizer, prompt, history=[]):
    chunk = response[output_pos:]
    output_pos = len(response)
    print(f"{chunk}", end="", flush=True)
```

输出：

```
笑脸的 emoji 是���,Unicode 编码为 U+263A。
```

### Expected Behavior

```
    笑脸的 emoji 是😊,Unicode 编码是 U+263A。
```

### Steps To Reproduce

```python
from transformers import AutoTokenizer, AutoModel, AutoModelForSeq2SeqLM

tokenizer = AutoTokenizer.from_pretrained("THUDM/chatglm-6b-int4-qe", trust_remote_code=True) #
model = AutoModel.from_pretrained("THUDM/chatglm-6b-int4-qe",trust_remote_code=True).float().eval()
```

    /home/a/miniconda3/lib/python3.10/site-packages/tqdm/auto.py:22: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
      from .autonotebook import tqdm as notebook_tqdm
    Explicitly passing a `revision` is encouraged when loading a model with custom code to ensure no malicious code has been contributed in a newer revision.
    Explicitly passing a `revision` is encouraged when loading a configuration with custom code to ensure no malicious code has been contributed in a newer revision.
    Explicitly passing a `revision` is encouraged when loading a model with custom code to ensure no malicious code has been contributed in a newer revision.

    No compiled kernel found.
    Compiling kernels : /home/a/.cache/huggingface/modules/transformers_modules/THUDM/chatglm-6b-int4-qe/92f16ed03b65b278a5d556ba2f4c18f03392d635/quantization_kernels_parallel.c
    Compiling gcc -O3 -fPIC -pthread -fopenmp -std=c99 /home/a/.cache/huggingface/modules/transformers_modules/THUDM/chatglm-6b-int4-qe/92f16ed03b65b278a5d556ba2f4c18f03392d635/quantization_kernels_parallel.c -shared -o /home/a/.cache/huggingface/modules/transformers_modules/THUDM/chatglm-6b-int4-qe/92f16ed03b65b278a5d556ba2f4c18f03392d635/quantization_kernels_parallel.so
    Kernels compiled : /home/a/.cache/huggingface/modules/transformers_modules/THUDM/chatglm-6b-int4-qe/92f16ed03b65b278a5d556ba2f4c18f03392d635/quantization_kernels_parallel.so
    Load kernel : /home/a/.cache/huggingface/modules/transformers_modules/THUDM/chatglm-6b-int4-qe/92f16ed03b65b278a5d556ba2f4c18f03392d635/quantization_kernels_parallel.so
    Setting CPU quantization kernel threads to 6
    Using quantization cache
    Applying quantization to glm layers

```python
prompt = "你知道笑脸的 emoji 是哪个吗？ Unicode编码是多少？"
response, history = model.chat(tokenizer, prompt, history=[])
print(response)
print("")

```

    The dtype of attention mask (torch.int64) is not bool

    笑脸的 emoji 是😊,Unicode 编码是 U+263A。
    

```python

prompt = "你知道笑脸的 emoji 是哪个吗？ Unicode编码是多少？"
output_pos = 0
for response, history in model.stream_chat(tokenizer, prompt, history=[]):
    chunk = response[output_pos:]
    output_pos = len(response)
    print(f"{chunk}", end="", flush=True)
```

    笑脸的 emoji 是���,Unicode 编码为 U+263A。

<img width="1179" alt="image" src="https://user-images.githubusercontent.com/12711404/230769448-857f689c-1e22-49e0-a16b-55be52fdfc0b.png">

### Environment

```markdown
- OS: ubuntu
- Python: 3.10
- Transformers: 4.27.4
- PyTorch: 2.0.0
- CUDA Support (`python -c "import torch; print(torch.cuda.is_available())"`) : cpu
```

**Comment by yunqiangwu** [comment]
解决了， 改成这样就没问题了

```
prompt = "你知道笑脸的 emoji 是哪个吗？ Unicode编码是多少？"
output_pos = 0
for response, history in model.stream_chat(tokenizer, prompt, history=[]):
    chunk = response[output_pos:]
    if "\uFFFD" == chunk[-1:]:
        continue
    output_pos = len(response)
    print(f"{chunk}", end="", flush=True)

```

<img width="799" alt="image" src="https://user-images.githubusercontent.com/26475774/230770888-9803f4f5-e5c3-4c9a-a270-f7aa83c1dbb4.png">

**Comment by yunqiangwu** [comment]
这个 issue 可以 close 了
